src/HTTP.jl

```diff
@@ -470,7 +478,8 @@ CodeInfo(:(begin
 end))
 ```
 """
-abstract type Layer end
+abstract type AbstractLayer end
+abstract type Layer{Next <: AbstractLayer} <: AbstractLayer end
```

Comment:
Do we need the `AbstractLayer` type?
While it lets us add a type constraint on `Next`
it also makes it easier for someone to mistakenly inherit from `AbstractLayer`
rather than `Layer`,